src/frontend/components/Chat/StudentChat/index.tsx

```diff
@@ -1,21 +1,75 @@
 import { Box } from 'grommet';
-import React from 'react';
+import React, { useState } from 'react';
+import { defineMessages, useIntl } from 'react-intl';
 
 import { ChatConversationDisplayer } from 'components/Chat/SharedChatComponents/ChatConversationDisplayer';
-import { ChatInputBar } from 'components/Chat/SharedChatComponents/ChatInputBar';
+import { InputBar } from 'components/Chat/SharedChatComponents/InputBar';
+import { InputDisplayNameOverlay } from 'components/Chat/SharedChatComponents/InputDisplayNameOverlay';
+import { JoinChatButton } from 'components/Chat/SharedChatComponents/JoinChatButton';
+import { converse } from 'utils/window';
 
 export const StudentChat = () => {
+  const [overlay, setOverlay] = useState(false);
+  const [isInputBarActive, setInputBarActive] = useState(false);
+  const intl = useIntl();
+
+  const processChatMessage = (chatMsg: string) => {
+    converse.sendMessage(chatMsg);
+    return true;
+  };
+
+  const messages = defineMessages({
+    inputBarPlaceholder: {
+      defaultMessage: 'Message...',
+      description:
+        'The input bar to write, edit and send messages to the chat conversation.',
+      id: 'components.InputBar.inputBarPlaceholder',
+    },
+  });
+
+  const handleJoinChatButton = () => {
+    setOverlay(true);
+  };
+
   return (
     <Box
-      direction="column"
       border={{
         color: 'blue',
         size: 'small',
       }}
+      direction="column"
       height="medium"
     >
-      <ChatConversationDisplayer />
-      <ChatInputBar />
+      <Box
+        style={{
+          display: overlay ? 'flex' : 'none',
+        }}
+      >
+        <InputDisplayNameOverlay
+          setInputBarActive={setInputBarActive}
+          setOverlay={setOverlay}
+        />
+      </Box>
+
+      <Box
+        height="100%"
+        style={{
+          display: overlay ? 'none' : 'flex',
```

Comment:
same question, even if here I think you need to mount the chat to connect to XMPP for now 

if this is the reason, you should comment it (and eventually create an issue to refer it and in other places that are in the same situation)

Reply:
I added a comment as you suggested. Not sure if it's necessary to open an issue or not, because I wil address this topic in a very next futur PR.